src/logging.c

```diff
@@ -135,12 +146,64 @@ void rcutils_logging_console_output_handler(
   }
 
   if (!location) {
-    fprintf(stream, "[%s] [%s]: %s\n", severity_string, name, message_buffer);
-  } else {
-    fprintf(
-      stream, "[%s] [%s]: %s (%s() at %s:%zu)\n", severity_string, name, message_buffer,
-      location->function_name, location->file_name, location->line_number);
+    // TODO before merge
+    //location = empty_location;
+  }
+
+  // Process the format string looking for known tokens
+  char token_start_delimiter = '{';
+  char token_end_delimiter = '}';
+  char output_buffer[1024];
+  memset(output_buffer, '\0', sizeof(output_buffer));
+  const char * str = g_rcutils_logging_output_format_string;
+  size_t size = strlen(g_rcutils_logging_output_format_string);
+
+  for (size_t i = 0; i < size; ++i) {
+    if (str[i] == token_start_delimiter) {
+      char token[20];
+      memset(token, '\0', sizeof(token));
+      size_t j = i + 1;
+      // Look for a token end delimeter
+      for (j = i + 1; j < size && str[j] != token_end_delimiter; j++);
+      if (j >= size) {
+          // No end delimiters found; there won't be any more tokens so shortcut the rest of the checking
+          strcat(output_buffer, str + i);
+          break;
+      }
+      size_t token_len = j - i - 1;  // not including delimeters
+      strncpy(token, str + i+1 , token_len);
+        char token_buffer[1024];
+        int n = 0;
+        bool known_token = false;
+        if (strcmp("severity", token) == 0)
+        {
+          n = sprintf(token_buffer, "%s", severity_string);
+          known_token = true;
+        }
+        if (!known_token && strcmp("line_number", token) == 0)
```

Comment:
for safety and performance `strncmp`

Reply:
Why would `strncmp` be "safer" here? What are you passing as `N`? The string literal first argument already limits the length to 12.

Reply:
you're right it is a string literal so the max length is already forced, habbit of using `strncmp` everywhere :S